Tensile/Source/lib/include/Tensile/msgpack/MessagePack.hpp

```diff
@@ -112,13 +113,19 @@ namespace Tensile
                 return MessagePackInput(otherObject, context);
             }
 
+            void addError(std::string const& msg)
+            {
+                error.push_back(msg);
+            }
+
             template <typename T>
             void mapRequired(const char* key, T& obj)
             {
-                auto map = objectToMap(object);
+                if(objectMap.empty())
+                    objectMap = objectToMap(object);
```

Comment:
Could just pass the map as reference second argument?  Would double check the assembly for this assignment

Reply:
Great suggestion, this is even faster.